src/main/java/org/terasology/gookeeper/system/HungerSystem.java

```diff
@@ -0,0 +1,203 @@
+
+        GooeyComponent gooeyComponent = gooeyEntity.getComponent(GooeyComponent.class);
+        HungerComponent hungerComponent = gooeyEntity.getComponent(HungerComponent.class);
+        HealthComponent healthComponent = gooeyEntity.getComponent(HealthComponent.class);
+
+        CharacterHeldItemComponent characterHeldItemComponent = event.getInstigator().getComponent(CharacterHeldItemComponent.class);
+
+        logger.info("Gooey Health: " + healthComponent.currentHealth);
+        if (characterHeldItemComponent != null && gooeyComponent.isCaptured) {
+            EntityRef item = characterHeldItemComponent.selectedItem;
+            String itemName = item.getComponent(DisplayNameComponent.class).name;
+
+            for (String acceptableFoodBlock : hungerComponent.foodBlockNames) {
+                if (!itemName.isEmpty() && itemName.equals(acceptableFoodBlock)) {
+                    delayManager.cancelPeriodicAction(gooeyEntity, Constants.healthDecreaseEventID);
+                    gooeyEntity.send(new GooeyFedEvent(event.getInstigator(), gooeyEntity, item));
+                    return;
+                }
+            }
+        }
+    }
+
+    /**
+     * Receives GooeyFedEvent when the targeted gooey is fed the held food block and hence resets the health to max.
+     *
+     * @param event,entity   The GooeyFedEvent, the gooey entity
+     */
+    @ReceiveEvent(components = {GooeyComponent.class})
+    public void onGooeyFed(GooeyFedEvent event, EntityRef entityRef) {
+        HealthComponent healthComponent = entityRef.getComponent(HealthComponent.class);
+        HungerComponent hungerComponent = entityRef.getComponent(HungerComponent.class);
+
+        healthComponent.currentHealth = healthComponent.maxHealth;
+        delayManager.addPeriodicAction(entityRef, Constants.healthDecreaseEventID, hungerComponent.timeBeforeHungry, hungerComponent.healthDecreaseInterval);
+
+        event.getItem().destroy();
+        entityRef.saveComponent(healthComponent);
+    }
+
+    /**
+     * Receives PeriodicActionTriggeredEvent when the gooey entity's health is decreased
+     *
+     * @param event,entity   The PeriodicActionTriggeredEvent, the gooey entity
+     */
+    @ReceiveEvent(components = {GooeyComponent.class})
+    public void onGooeyHealthDecrease(PeriodicActionTriggeredEvent event, EntityRef entity) {
+        if (event.getActionId().equals(Constants.healthDecreaseEventID)) {
+            HealthComponent healthComponent = entity.getComponent(HealthComponent.class);
+            HungerComponent hungerComponent = entity.getComponent(HungerComponent.class);
+
+            healthComponent.currentHealth -= hungerComponent.healthDecreaseAmount;
+            entity.saveComponent(healthComponent);
+
+            if (healthComponent.currentHealth <= 0) {
+                entity.destroy();
+            }
+        }
+    }
+
+    @ReceiveEvent(components = {GooeyComponent.class, HungerComponent.class})
```

Comment:
If you have the components as arguments in thr method, you don't need them in the annotation here.